src/QuadForm/Lattices.jl

```diff
@@ -953,6 +965,12 @@ is_isotropic(L::AbsLat, p) = is_isotropic(rational_span(L), p)
 #
 ################################################################################
 
+@doc Markdown.doc"""
+    restrict_scalars(L::AbsLat) -> ZLat
+
+Given a lattice `L`, return the $\mathbb Z$-lattice obtained by restriction of
+scalars.
```

Comment:
Note that restriction of scalars for a quadratic_space has a different signature
```
restrict_scalars(V::AbsSpace, K::FlintRationalField,
                                       alpha = one(base_ring(V)))
```
Perhaps
```
restrict_scalars(V::AbsLat, K::FlintRationalField,
                                       alpha = one(base_ring(V)))
```
would be the better signature for this function?

Reply:
Could you please include a docstring for 
```
restrict_scalars(V::AbsSpace, K::FlintRationalField,
                                       alpha = one(base_ring(V)))
```

Reply:
I guess it depends on what we want to do with restriction of scalars for `AbsLat`... Are we willing to leave to choice not to restrict to `QQ` but only a subfield of the `base_field` ? Is it relevant in practice ? If yes then I could make the change also for the one returning the map.

Reply:
It is not relevant for me but it could be relevant for others. Thus I like the syntax. It allows to add a restriction of scalars for subfields whenever we need to.
Please also add `saturate(X::AbsLat,Y::AbsLat) = primitive_closure(X, Y)`

Reply:
Yes but now that I look more at it, I realized that: `K::FlintRationalField` is part of the signatures but we don't have the choice... we can only have `QQ` for this. So in a sense, we put `K` as argument for where we restrict the scalars, but at the end of the day, we restrict everything, and everytime, to `QQ`

Reply:
Yes, the point is that in the future we can add a method for restriction to other fields without changing the syntax
and by simply writing a new method.  
We could allow `restriction(V::AbsSpace, alpha=...)` for restriction to `QQ` though. Any opinion @thofma ?

Reply:
I guess in general one would need to do a total change on `VecSpaceRes` and `SpaceRes` to allow restriction of scalars to smaller but non rational field, and then starts to change all the functions using restriction of scalars accoridngly. Could it be worth to dedicate a proper PR for this ?

Reply:
I would not implement it for now because we don't need it. But maybe someone will need it in the future. You never know.

Reply:
I am not too big of a fan of `restriction(V::AbsSpace,...)`, as this could mean other things. Also I think either there is a function argument making it clear that one restricts to the rationals or it is indicated in the name.

For lattices, would you restrict scalars to QQ or to ZZ? So what about

```
restrict_scalars(V::AbsLat, K::FlintIntegerRing,
                                       alpha = one(base_ring(V)))
```

It is a bit annoying, since one will have to explain the `alpha` again.

Reply:
I have added appropriate documentation and made things consistent with the restriction of scalars for `AbsSpace`. I have left open the idea of extending this to any subfield of the base algebra there, so that users know what is available, and if someone needs it one day, it can be implemented without changing too much the format (maybe adapting the documentation and all the attached functions)